ext/node/resolution.rs

```diff
@@ -39,53 +46,1260 @@ impl NodeResolutionMode {
   }
 }
 
-/// Checks if the resolved file has a corresponding declaration file.
-pub fn path_to_declaration_path<Fs: NodeFs>(
-  path: PathBuf,
-  referrer_kind: NodeModuleKind,
-) -> Option<PathBuf> {
-  fn probe_extensions<Fs: NodeFs>(
-    path: &Path,
+#[derive(Debug)]
```

Comment:
The changes to this file were mostly just combining everything from resolver.rs into here. I didn't do any logic changes. We can keep improving this in the future.